### PR TITLE
New protocol: Movie Dose analysis

### PR DESCRIPTION
**Pull request by danielmarchan3**
- It estimates the mean dose per angstrom**2 to check how stable is the dose during the acquisition. 
- This can be useful for facilities

### CODE REVIEW

File: xmipp3/protocols/protocol_movie_dose_analysis.py [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+
+def plotDoseAnalysisDiff(filename, medianDifferences):
+    medianDiff = np.median(medianDifferences)
+    x = np.arange(start=1+1, stop=len(medianDifferences)+2, step=1)
+    plt.figure()
+    plt.scatter(x, medianDifferences, s=10)
+    plt.axhline(y=medianDiff, color='r', linestyle='-',  label='Median dose difference')
+    plt.xlabel("Movies ID")
+    plt.ylabel("Dose differences")
+    plt.title('Dose differences with respect to the global median vs time')
```
Comment (albertmena) [inline xmipp3/protocols/protocol_movie_dose_analysis.py:397]: Add the units (e⁻/A²) ?

File: xmipp3/protocols/protocol_movie_dose_analysis.py [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+             'min': min_dose,
+             }
+    return stats
+
+def plotDoseAnalysis(filename, doseValues, medianGlobal, lower, upper):
+    x = np.arange(start=1, stop=len(doseValues)+1, step=1)
+    plt.figure()
+    plt.scatter(x, doseValues,s=10)
+    plt.axhline(y=medianGlobal, color='r', linestyle='-', label='Median dose')
+    plt.axhline(y=upper, color='b', linestyle='-.', label='Upper limit')
```
Comment (albertmena) [inline xmipp3/protocols/protocol_movie_dose_analysis.py:380]: Upper limit **dose**
Lower limit **dose**

**Review by albertmena — CHANGES_REQUESTED** [review]
Looks great and very useful for facilities!, just some details to fix. Are you planning to run this protocol in streaming?

File: xmipp3/protocols/protocol_movie_dose_analysis.py [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+from pwem.objects import SetOfMovies
+from pwem.protocols import ProtProcessMovies
+from xmipp3.convert import getScipionObj
+import statistics as stat
+
+THRESHOLD = 0.02
+
+class XmippProtMovieDoseAnalysis(ProtProcessMovies):
+    """ Protocol for the dose analysis """
+    # FIXME: WITH .mrcs IT DOES NOT FILL THE LABELS
```
[the remark this answers is not in the document]
Comment (danielmarchan3) [reply xmipp3/protocols/protocol_movie_dose_analysis.py:46]: Work in progress, not sure why I am having problems with this type of movies format

**Review by albertmena — APPROVED** [review]

**Pull request merged**